py3status/modules/wwan.py

```diff
@@ -0,0 +1,563 @@
+            'format_ipv6', 'format_stats', 'format_message', 'message*'
+        ]
+        format_strings = [self.format, self.format_notification]
+        for name, placeholder in zip(names, placeholders):
+            self.init[name] = []
+            for format_string in format_strings:
+                if format_string:  # fails on None, [], '', etc
+                    if self.py3.format_contains(format_string, placeholder):
+                        self.init[name].append(format_string)
+                        if name in ['ipv4', 'ipv6']:
+                            if name not in self.init['ip']:
+                                self.init['ip'].append(name)
+                        if name in ['message', 'format_message']:
+                            if name not in self.init['sms_message']:
+                                self.init['sms_message'].append(name)
+
+    def _get_modem_proxy(self):
+        modemmanager_proxy = self.bus.get(STRING_MODEMMANAGER_DBUS)
+        modems = modemmanager_proxy.GetManagedObjects()
+
+        for objects in modems.items():
+            modem_path = objects[0]
+            modem_proxy = self.bus.get(STRING_MODEMMANAGER_DBUS, modem_path)
+            eqid = modem_proxy.EquipmentIdentifier
+
+            if self.modem is None or self.modem == eqid:
+                return modem_proxy
+        else:
+            return {}
+
+    def _get_modem_status_data(self, modem_proxy):
+        modem_data = {}
+        try:
+            modem_data = modem_proxy.GetStatus()
+        except:
+            pass
+        return modem_data
+
+    def _get_bearer(self, modem_proxy):
+        bearer = {}
+        try:
+            bearer = modem_proxy.Bearers[0]
+        except:
+            pass
+        return bearer
+
+    def _get_interface(self, bearer):
+        return self.bus.get(STRING_MODEMMANAGER_DBUS, bearer).Interface
+
+    def _get_message_data(self, modem_proxy):
+        message_data = {}
+        try:
+            message_data = modem_proxy.Messages
+        except:
+            pass
+        return message_data
+
+    def _count_messages(self, message_data):
+        count_messages = len(message_data)
+        count_message = max(0, count_messages - self.last_messages)
+        self.last_messages = count_messages
+        return count_message, count_messages
+
+    def _manipulate_message(self, data):
+        new_message = []
+        for index, msg in sorted(enumerate(data, 1), reverse=True):
+            try:
+                sms_proxy = self.bus.get(STRING_MODEMMANAGER_DBUS, msg)
+                new_message.append(
+                    self.py3.safe_format(
+                        self.format_message, {
+                            'index': index,
+                            'id': msg.rsplit('/', 1)[-1],
+                            'number': sms_proxy.Number,
+                            'text': sms_proxy.Text,
+                            'timestamp': sms_proxy.Timestamp,
```

Comment:
This placeholder is unlisted. I guess we should remove this. Probably not much of use to anyone. :-(

Reply:
Timestamp, but yes, I suppose id too. I guess it's not that useful and there are too many garbage data to expose them all.